### ISSUE TITLE
Bug: prismjs is included as a dependency of @lexical/markdown

### ISSUE THREAD

**Issue by markulrich** [facebook/lexical#5381]
It appears that prismjs gets included as a (very large) dependency of @lexical/markdown, swelling the bundle size.

## Steps To Reproduce

See https://bundlephobia.com/package/@lexical/markdown@0.12.5 and that prismjs swells the bundle size

## The current behavior

Prismjs is included in the bundle

## The expected behavior

Prismjs is not included, and the bundle size is <100kb

**Comment by markulrich** [comment]
Also see note from @acywatson at https://github.com/facebook/lexical/discussions/3778#discussioncomment-6694006. Thank you!!

**Comment by vadimkantorov** [comment]
Btw, just curious, what is the expected `ls node_modules | wc -l` after `npm install` of a freshly-cloned https://github.com/facebook/lexical ?

For me this initial `npm install` led to more than 1000...

**Comment by etrepum** [comment]
I think this is really just bundlephobia telling you what the worst case is, if you were to use every single thing with no tree-shaking. Can you demonstrate that this actually happens in practice, with a real bundler in an application that is not using code nodes?

**Comment by vadimkantorov** [comment]
No, but I am also not a web developer and now knowledgeable of proper use of bundlers etc. So the mere number of fetched modules scared me and I was wondering what are some typical bundle size and if I'm doing anything wrong with this 1000 of modules...

**Comment by etrepum** [comment]
I was commenting on the original post, not your off-topic reply.

**Comment by AlessioGr** [comment]
Including Prism in @lexical/markdown caused a messed up issue for us where Node would randomly exit the process if @lexical/markdown was imported dynamically (or any dynamically imported module in the chain imported from @lexical/markdown). Getting rid of the `@lexical/code` import fixed it for us: https://github.com/payloadcms/payload/pull/9382.

Wondering if anyone else has experienced this issue

**Comment by etrepum** [comment]
I'm sure that could happen depending on how your bundler omits modules and what kind of exception handling your code has, but the anecdote isn't really of much use without a repro. Of course the best thing to do would be to refactor things accordingly to make prism explicitly optional, the next best thing would be to demonstrate a working bundler configuration that excludes the majority of prism (e.g. the languages).

**Comment by AlessioGr** [comment]
This issue actually occurred outside of a bundler. It was a straightforward script executed using node, tsx or swc. Interestingly, it worked perfectly within a bundler. The problem is frustratingly difficult to reproduce, as it only occurred about 0.1% of the time and seemed dependent on how _much_ code was being dynamically imported, and no error was thrown.

Spent quite some time debugging this issue with 0 results online, so I had to share it hah. Adds some more weight to the argument for removing this dependency

**Comment by etrepum** [comment]
Looks like the sort of mistake that vscode will create, I doubt it’s intentional

**Comment by AlessioGr** [comment]
I think this code was written [before those exports were exported from `@lexical/utils`](https://github.com/facebook/lexical/pull/5831). Some modules were manually copy and pasted from the shared package. Can definitely be updated now
